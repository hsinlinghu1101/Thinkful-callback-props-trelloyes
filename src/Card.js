import React from 'react';
import './Card.css'

export default function  Card (props){
 
        return(

            <div key={props.id} className="Card" >
                <button  onClick={() => props.onDelete(props.id)} type="button">delete</button>
                <h3>{props.title}</h3>
                <p>{props.content}</p>
              </div>
        )

}


