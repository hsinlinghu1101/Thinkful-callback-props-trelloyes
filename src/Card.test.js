
import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
//snapshot test
import renderer from 'react-test-renderer';


/* smoke test
it ('renders learn react link', () => {
  
   const main =document.createElement('div')
   ReactDOM.render(<Card/>, main)
   ReactDOM.unmountComponentAtNode(main);
});*/

it ('renders learn react link', () => {
  
    const main =document.createElement('div')
    ReactDOM.render(<Card />, main)
    ReactDOM.unmountComponentAtNode(main);
 });

 it('render snapshot test', () =>{
     const tree = renderer
     
    .create(<Card  key='test'
    title='test'
    content='test'/>).toJSON();
    expect(tree).toMatchSnapshot();
 })
