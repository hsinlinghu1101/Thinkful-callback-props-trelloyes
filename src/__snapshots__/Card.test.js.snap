// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot test 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;
